ethereumj-core/src/main/java/org/ethereum/solidity/SolidityType.java

```diff
@@ -410,6 +412,48 @@ public static BigInteger decodeInt(byte[] encoded, int offset) {
         public static byte[] encodeInt(BigInteger bigInt) {
             return ByteUtil.bigIntegerToBytesSigned(bigInt, 32);
         }
+        @Override
+        public Object decode(byte[] encoded, int offset) {
+            return decodeInt(encoded, offset);
+        }
+        @Override
+        public byte[] encode(Object value) {
+            BigInteger bigInt = encodeInternal(value);
+            return encodeInt(bigInt);
+        }
+    }
+    
+    public static class UnsignedIntType extends NumericType {
+        public UnsignedIntType(String name) {
+            super(name);
+        }
+        
+        @Override
+        public String getCanonicalName() {
+            if (getName().equals("uint")) return "uint256";
+            return super.getCanonicalName();
+        }
+        public static BigInteger decodeInt(byte[] encoded, int offset) {
+            return new BigInteger(1, Arrays.copyOfRange(encoded, offset, offset + 32));
+        }
+        public static byte[] encodeInt(int i) {
+            return encodeInt(new BigInteger("" + i));
```

Comment:
One more check here for i >= 0

Reply:
ahhh, it will go through another one, my miss

Reply:
This one isn't necessary as `encodeInt(int i)` uses `static encodeInt(BigInteger bigInt)` where I placed the check. Insert anyway?